lib/task_listener.js

```diff
@@ -229,7 +227,18 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
-  addRunningTask(runningState) {
+  async recordCapacity () {
+    this.runtime.stats.record('capacityOverTime', {
+      duration: Date.now() - this.timestamp,
+      availableCapacity: await this.availableCapacity(),
```

Comment:
could just set something like lastKnownCapacity each time availableCapacity is called, and the stats recording will just use lastKnownCapacity